src/ecs/rendering/systems.rs

```diff
@@ -49,5 +56,135 @@ where
 
         self.renderer
             .draw(&mut self.pipe, data, Duration::from_secs(0));
+
+        let mut events: Vec<Event> = Vec::new();
+        self.renderer.events_mut().poll_events(|new_event| {
+            compress_events(&mut events, new_event);
+        });
+
+        if let Err(err) = event_handler.slice_write(&events) {
+            eprintln!(
+                "WARNING: Writing too many window events this frame! {:?}",
+                err
+            );
+        }
+    }
+}
+
+/// Input devices can sometimes generate a lot of motion events per frame, these are
+/// useless as the extra precision is wasted and these events tend to overflow our
+/// otherwise very adequate event buffers.  So this function removes and compresses redundant
+/// events.
+fn compress_events(vec: &mut Vec<Event>, new_event: Event) {
```

Comment:
We could have that in a generic input crate some day.

Reply:
No, I think compression should be as close to the source as possible to not flood the system with unnecessary events.